File: p3.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func timeConversion(input_time string) string {
	t := strings.Split(input_time[:len(input_time)-2], ":")
	ap := input_time[len(input_time)-2:]
	h, _ := strconv.Atoi(t[0])
	if ap == "PM" && h != 12 {
		t[0] = strconv.Itoa(h + 12)
	} else if ap == "AM" && h == 12 {
		t[0] = "00"
	}

	return strings.Join(t, ":")
}

func main() {
	fmt.Println("convert:", timeConversion("12:45:54PM"))
}

File: p3.py
def timeConversion(s):
    time = s[0:-2].split(':')
    ampm = s[-2:]
    if ampm == 'PM' and time[0] != '12':
        time[0] = str(int(time[0]) + 12)
    elif time[0] == '12':
        time[0] = '00'

    return ':'.join(time)


print(timeConversion('12:01:00PM'))